plugins/tiledb/test/XYZTmUtils.cpp

```diff
@@ -0,0 +1,220 @@
+/******************************************************************************
+* Copyright (c) 2021 TileDB, Inc.
+*
+* All rights reserved.
+*
+* Redistribution and use in source and binary forms, with or without
+* modification, are permitted provided that the following
+* conditions are met:
+*
+*     * Redistributions of source code must retain the above copyright
+*       notice, this list of conditions and the following disclaimer.
+*     * Redistributions in binary form must reproduce the above copyright
+*       notice, this list of conditions and the following disclaimer in
+*       the documentation and/or other materials provided
+*       with the distribution.
+*     * Neither the name of Hobu, Inc. or Flaxen Geo Consulting nor the
+*       names of its contributors may be used to endorse or promote
+*       products derived from this software without specific prior
+*       written permission.
+*
+* THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+* "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+* LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+* FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+* COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+* INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+* BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+* OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+* AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+* OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT
+* OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY
+* OF SUCH DAMAGE.
+****************************************************************************/
+
+#include "XYZTmUtils.hpp"
+#include <pdal/Metadata.hpp>
+
+
+namespace pdal
+{
+
+static StaticPluginInfo const s_info
+{
+    "readers.XYZTimeFauxReader",
+    "XYZ time Faux Reader",
+    "none"
+};
+
+CREATE_STATIC_STAGE(XYZTimeFauxReader, s_info);
+
+std::string XYZTimeFauxReader::getName() const
+{
+    return s_info.name;
+}
+
+void XYZTimeFauxReader::addArgs(ProgramArgs& args)
+{
+    args.add("bounds", "X/Y/Z/time limits",
+                 m_bounds, BOX4D(0., 0., 0., 0.,
+                                        1., 1., 1., 1));
+    args.add("xyz_mode", "mode for distribution of xyz dimension values", m_xyz_mode, Mode::Uniform);
+    args.add("time_mode", "mode for distribution of time dimension values", m_tm_mode, Mode::Ramp);
+    args.add("use_time", "Add a time dimension in addition to XYZ (default true)", m_use_time, true);
+    args.add("density", "Double value to set density dimension in points", m_density, 1.0);
+}
+
+void XYZTimeFauxReader::prepared(PointTableRef table)
+{
+    if (!m_countArg->set())
```

Comment:
Maybe I'm missing something but this doesn't seem to inherit from FauxReader and I don't see `m_count` or `m_countArg` defined.

Reply:
They're inherited from the Reader class. I saw the FauxReader set up like that so I just kind of used that pattern.